projects/combined_sequences/generate_plots.py

```diff
@@ -0,0 +1,363 @@
+
+  ########################################################################
+  #
+  # Create the plot.
+  plt.figure()
+  plotPath = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                          "plots", "sensorimotorAccuracy_by_sequence.pdf")
+
+  # Plot each curve
+  legendList = []
+  colorList = ['r', 'b', 'g', 'm', 'c', 'k', 'y']
+
+  for i in range(len(featureRange)):
+    f = featureRange[i]
+    print "features={} locationRange={} accuracy={}".format(
+      f,locationRange, accuracy[f,locationRange]),
+    print totals[f,locationRange]
+    legendList.append('Sensorimotor layer, feature pool size: {}'.format(f))
+    plt.plot(locationRange, accuracy[f,locationRange],
+             color=colorList[i])
+
+  plt.plot(locationRange, [100] * len(locationRange),
+           color=colorList[len(featureRange)])
+  legendList.append('Temporal sequence layer')
+
+  # format
+  plt.legend(legendList, bbox_to_anchor=(0., 0.65, 1., .102), loc="right", prop={'size':10})
+  plt.xlabel("Size of location pool")
+  # plt.xticks(range(0,max(locationRange)+1,10))
+  # plt.yticks(range(0,int(accuracy.max())+2,10))
+  plt.ylim(-10.0, 110.0)
+  plt.ylabel(yaxis)
+  plt.title(title)
+
+    # save
+  plt.savefig(plotPath)
+  plt.close()
+
+
+
+
+if __name__ == "__main__":
+
+  dirName = os.path.dirname(os.path.realpath(__file__))
+
+  parser = OptionParser("python %prog [option ...]")
+  parser.add_option("--fig4A",
+                    default=False, action="store_true", help="Plot Fig 4A.")
+  parser.add_option("--fig4B",
+                    default=False, action="store_true", help="Plot Fig 4B.")
+  parser.add_option("--fig5A",
+                    default=False, action="store_true", help="Plot Fig 5A.")
+  parser.add_option("--fig5B",
+                    default=False, action="store_true", help="Plot Fig 5B.")
+  parser.add_option("--fig6",
+                    default=False, action="store_true", help="Plot Fig 6.")
+
+  # Parse CLI arguments
+  options, args = parser.parse_args(sys.argv[1:])
+
+
+  # Generate images similar to those used in the first plot for the section
+  # "Simulations with Pure Temporal Sequences"
+  if options.fig4A:
+    resultsFilename = os.path.join(dirName, "pure_sequences_example.pkl")
+    with open(resultsFilename, "rb") as f:
+      results = cPickle.load(f)
+
+    for trialNum, stat in enumerate(results["statistics"]):
+      plotOneInferenceRun(
+        stat,
+        itemType="a single sequence",
+        fields=[
+          ("L4 PredictedActive", "Predicted active cells in sensorimotor layer"),
+          ("TM NextPredicted", "Predicted cells in temporal sequence layer"),
+          ("TM PredictedActive",
+           "Predicted active cells in temporal sequence layer"),
+        ],
+        basename="pure_sequences",
+        trialNumber=trialNum,
+        plotDir=os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                             "detailed_plots")
```

Comment:
Why the 2 different plots folders (`plots` vs `detailed_plots`)?

Reply:
The `detailed_plots` folder contains tons of individual plots.